### ISSUE TITLE
TypeError: Cannot read property 'constructor' of undefined

### ISSUE THREAD

**Issue by tomvolek** [meteor/meteor#10562]
* ❗️ TypeError: Cannot read property 'constructor' of undefined 

**Meteor version:**  latest build : 1.8.1  
**OS:**  Mac OS Mojave 
**Description:** After upgrading from Meteor 1.7 to 1.8.1 I get the error "TypeError: Cannot read property 'constructor' of undefined"  . Few packages listed at the end can be upgraded individually, but I have not done that yet to see if it makes any difference. 
Stack trace is as follows: 
**meteor run** 
=> Started proxy.                             
=> Started MongoDB.                           
W20190516-13:03:46.132(-7)? (STDERR) /Users/tom/.meteor/packages/meteor-tool/.1.8.1.7c0itz.u7ktr++os.osx.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.osx.x86_64/dev_bundle/server-lib/node_modules/fibers/future.js:280
W20190516-13:03:46.173(-7)? (STDERR) 						throw(ex);
W20190516-13:03:46.173(-7)? (STDERR) 						^
W20190516-13:03:46.173(-7)? (STDERR) 
W20190516-13:03:46.174(-7)? (STDERR) TypeError: Cannot read property 'constructor' of undefined
W20190516-13:03:46.174(-7)? (STDERR)     at exposeSubscription (packages/meteorhacks_meteorx.js:61:41)
W20190516-13:03:46.174(-7)? (STDERR)     at exposeLivedata (packages/meteorhacks_meteorx.js:31:5)
W20190516-13:03:46.174(-7)? (STDERR)     at packages/meteorhacks_meteorx.js:175:1
W20190516-13:03:46.175(-7)? (STDERR)     at packages/meteorhacks_meteorx.js:200:4
W20190516-13:03:46.175(-7)? (STDERR)     at packages/meteorhacks_meteorx.js:208:3
W20190516-13:03:46.175(-7)? (STDERR)     at /Users/tom/Development/payagol/.meteor/local/build/programs/server/boot.js:419:36
W20190516-13:03:46.175(-7)? (STDERR)     at Array.forEach (<anonymous>)
W20190516-13:03:46.175(-7)? (STDERR)     at /Users/tom/Development/payagol/.meteor/local/build/programs/server/boot.js:228:19
W20190516-13:03:46.175(-7)? (STDERR)     at /Users/tom/Development/payagol/.meteor/local/build/programs/server/boot.js:479:5
W20190516-13:03:46.176(-7)? (STDERR)     at Function.run (/Users/tom/Development/payagol/.meteor/local/build/programs/server/profile.js:510:12)
W20190516-13:03:46.176(-7)? (STDERR)     at /Users/tom/Development/payagol/.meteor/local/build/programs/server/boot.js:478:11
=> Exited with code: 1

**Packages installed are :** 
Tom2s-MacBook-Pro:payagol tom$ meteor list
accounts-base                                 1.4.4  A user account system
accounts-password                             1.5.1  Password support for accounts
alanning:roles                                1.2.16  Authorization package for Meteor
aldeed:autoform                               6.3.0  Easily create forms with automatic insert and update, and automatic reactive validation.
aldeed:collection2-core                       2.1.2  Core package for aldeed:collection2
alexwine:pdfmake                              0.1.26_8  Let's you generate PDF's on the client and provides handy features for Layout & Styling.
anti:i18n                                     0.4.3  Internalization: simplest package
aslagle:reactive-table                        0.8.45  A reactive table designed for Meteor
blaze-html-templates                          1.1.2  Compile HTML templates into reactive UI with Meteor Blaze
cfs:standard-packages                         0.5.10  Filesystem for Meteor, collectionFS
check                                         1.3.1  Check whether a value matches a pattern
clinical:barcode                              3.1.0  Provides the 3 of 9 Barcode font.
danappelxx:hover.css                          1.0.0  A wrapper for hover.css to add smooth css animations to your pages.
dburles:collection-helpers                    1.1.0  Transform your collections with helpers that you define
ddp                                           1.4.0  Meteor's latency-compensated distributed data framework
ddp-client                                    2.3.3  Meteor's latency-compensated distributed data client
ddp-server                                    2.3.0  Meteor's latency-compensated distributed data server
dynamic-import                                0.5.1  Runtime support for Meteor 1.5 dynamic import(...) syntax
ecmascript                                    0.12.7  Compiler plugin that supports ES2015+ in all .js files
ejson                                         1.1.0  Extended and Extensible JSON library
email                                         1.2.3  Send email messages
es5-shim                                      4.8.0  Shims and polyfills to improve ECMAScript 5 support
facts                                         1.0.9  Publish internal app statistics
fortawesome:fontawesome                       4.7.0  Font Awesome (official): 500+ scalable vector icons, customizable via CSS, Retina friendly
fullcalendar:fullcalendar                     2.9.0  Full-sized drag & drop event calendar
gadicohen:messageformat                       0.0.53  MessageFormat support, the Meteor way
http                                          1.4.2  Make HTTP calls to remote servers
ian:accounts-ui-bootstrap-3                   1.2.89  Bootstrap-styled accounts-ui with multi-language support.
ignacy130:yadmin                              2.0.0  Simple and easy to use admin panel, continuation of yogiben-meteor:admin. Works with Meteor 1.5.
insecure                                      1.0.7  (For prototyping only) Allow all database writes from the client
iron:middleware-stack                         1.1.0  Client and server middleware support inspired by Connect.
iron:router                                   1.1.2  Routing specifically designed for Meteor
jquery                                        1.11.11  Manipulate the DOM using CSS selectors
lamhieu:meteorx                               2.1.1  Exposing internal Meteor APIs to hack Meteor easily
lc3t35:datatables-bootstrap3                  1.10.10  Bootstrap'ed DataTables for meteor
less                                          2.8.0  Leaner CSS language
logging                                       1.1.20  Logging facility.
maazalik:highcharts                           0.4.0  HighCharts for Meteor, with an easy to use helper to get you started!
meteor-base                                   1.4.0  Packages that every Meteor app needs
minimongo                                     1.4.5  Meteor's client-side datastore: a port of MongoDB to Javascript
mizzao:bootboxjs                              4.4.0  Programmatic dialog boxes using Twitter's bootstrap modals
mizzao:timesync                               0.5.1  NTP-style time synchronization between server and client
mizzao:user-status                            0.6.8  User connection and idle state tracking for Meteor
mobile-experience                             1.0.5  Packages for a great mobile user experience
momentjs:moment                               2.24.0  Moment.js (official): parse, validate, manipulate, and display dates - official Meteor packaging
mongo                                         1.6.2  Adaptor for using MongoDB and Minimongo over DDP
multiply:iron-router-progress                 1.0.2  Progressbar for iron:router
ostrio:files                                  1.10.2  File upload via DDP/HTTP to server, 3rd party storage support: AWS S3, GridFS, DropBox and others
peerlibrary:stacktrace                        1.3.1_2  Meteor package for getting uniform stack traces in all web browsers and server
peppelg:bootstrap-3-modal                     1.0.4  Simple usage of bootstrap 3 modals.
phowat:i18n-ui-dropdown-bootstrap3-component  0.0.1  Replacement for tap-i18n-ui's i18n_dropdown as a Bootstrap 3 dropdown component.
rajit:bootstrap3-datepicker                   1.7.1_1  Meteor packaging of uxsolutions/bootstrap-datepicker for Bootstrap 3
rajit:bootstrap3-datepicker-fa                1.7.1  Meteor packaging of Persian translation of uxsolutions/bootstrap-datepicker
random                                        1.1.0  Random number generator and utilities
react-meteor-data                             0.2.16  React higher-order component for reactively tracking Meteor data
react-template-helper                         0.2.11  Use React components in native Meteor templates
reactive-var                                  1.0.11  Reactive variable
reload                                        1.3.0  Reload the page while preserving application state.
rocketchat:streamer                           0.6.2* DB less realtime communication for meteor
rzymek:moment-locale-fa                       2.14.1  Moment.js fa locale. Companion package for rzymek:moment
selaias:avatar-manager                        0.4.0  Manage your avatar. Upload an imag or select one from the available list (linked accounts).
session                                       1.2.0  Session variable
shell-server                                  0.4.0  Server-side component of the `meteor shell` command.
showdown                                      1.0.8  Moved to the 'markdown' package
spacebars                                     1.0.15  Handlebars-like template language for Meteor
standard-minifier-css                         1.5.3  Standard css minifier used with Meteor apps by default.
standard-minifier-js                          2.4.1  Standard javascript minifiers used with Meteor apps by default.
tap:i18n                                      1.8.2  A comprehensive internationalization solution for Meteor
tap:i18n-db                                   0.4.0  Internationalization for Meteor Collections
tap:i18n-ui                                   0.8.0  User interface for the tap-i18n package
telerik:kendo-ui-core-bootstrap-theme         2015.2.805  Kendo UI Core is the free & open-source version of Kendo UI, the web's best UI widgets!
themeteorchef:bert                            2.2.0  A client side, multi-style alerts system for Meteor.
tmeasday:presence                             1.0.6  A package to help track users' presence
tomi:upload-jquery                            2.4.0  Client template for uploads using "jquery-file-upload" from blueimp
tomi:upload-server                            1.3.4_3  Upload server for Meteor. Allows to save and serve files from arbitrary directory
tracker                                       1.2.0  Dependency tracker to allow reactive callbacks
twbs:bootstrap                                3.3.6  The most popular front-end framework for developing responsive, mobile first projects on the web.
underscore                                    1.0.10  Collection of small helpers: _.map, _.each, ...
useraccounts:iron-routing                     1.14.2  UserAccounts package providing routes configuration capability via iron:router.
utilities:avatar                              0.9.2  Consolidated user avatar template (twitter, facebook, gravatar, etc.)
webapp                                        1.7.4  Serves a Meteor app over HTTP
wenzhixin:bootstrap-table                     1.9.0  Bootstrap table with radio, checkbox, sort, pagination, and other added features.
xvendo:barcoder                               0.0.3  Easily generate barcodes
yuukan:streamy                                1.4.1  Simple interface to use the underlying sockjs in a meteor application
zardak:soap                                   0.2.6  Wrapped npm soap package

                                              
* New versions of these packages are available! Run 'meteor update' to try to update those packages to their latest versions. If your packages cannot be
  updated further, try typing `meteor add <package>@<newVersion>` to see more information.

**meteor update**  
This project is already at Meteor 1.8.1, the latest release.
Your top-level dependencies are at their latest compatible versions.
                                              
The following top-level dependencies were not updated to the very latest version available:
 * rocketchat:streamer 0.6.2 (1.0.1 is available)
                                              
Newer versions of the following indirect dependencies are available:
 * aldeed:tabular 1.6.1 (2.1.1 is available)  
 * coffeescript 1.0.17 (2.3.2_1 is available) 
 * fourseven:scss 4.10.0 (4.12.0 is available)
 * raix:eventemitter 0.1.3 (1.0.0 is available)
 * reywood:publish-composite 1.5.2 (1.7.0 is available)
 * softwarerero:accounts-t9n 1.3.11 (2.5.1 is available)
These versions may not be compatible with your project.
To update one or more of these packages to their latest
compatible versions, pass their names to `meteor update`,
or just run `meteor update --all-packages`.

**Comment by alihuber** [comment]
Duplicate of / possibly related to https://github.com/meteor/meteor/issues/10553, https://github.com/meteorhacks/meteorx/issues/8 and https://github.com/meteor/meteor/issues/10440 ?

**Comment by tomvolek** [comment]
alihuber: I have tried the suggested actions in the above threads , none worked. 
I am running Meteor on MacOSX and have installed https://github.com/lamhieu-vk/meteorx 
I also tried suggestion of setting up a proxy package and still not working.

**Comment by zodern** [comment]
Try running `meteor list --tree` and look for a package using `meteorhacks:meteorx`.

**Comment by tomvolek** [comment]
zodern:    Meteor list-tree shows followings,do i remove  meteorhacks:meteorx  ?  Trying to meteor remove meteorhacks:meteorx  says: meteorhacks:meteorx is not a direct dependency in this project.

─┬ meteorhacks:unblock@1.1.0
│ └─┬ meteorhacks:meteorx@1.4.1
│   ├── mongo@1.6.2 (top level)
│   └── random@1.1.0 (top level)

I have added this package per some comments in google threads. 
lamhieu:meteorx@2.1.1
├── mongo@1.6.2 (top level)
└── random@1.1.0 (top level)

**Comment by maxfi** [comment]
@tomvolek You need to replace the `meteorhacks:unblock` package with the `lamhieu:unblock` package. To do so run:

```sh
meteor remove meteorhacks:unblock
meteor add lamhieu:unblock
```

You don't need to explicitly define `lamhieu:meteorx` as from your `meteor list --tree` output the only dependent package on `meteorx` was `meteorhacks:unblock`. So you can remove `lamhieu:meteorx` from your `packages` file.

**Comment by tomvolek** [comment]
I did as mentioned above,  but I get the following 👍 meteor remove  meteorhacks:unblock
meteorhacks:unblock is not a direct dependency in this project.

--------
meteor list --tree  | grep meteorhacks
│ ├─┬ meteorhacks:inject-initial@1.0.4
├── meteorhacks:inject-initial@1.0.4 (expanded above)
├─┬ meteorhacks:unblock@1.1.0
│ └─┬ meteorhacks:meteorx@1.4.1
├── meteorhacks:inject-initial@1.0.4 (expanded above)
----------
meteor list --tree  | grep lamhieu
lamhieu:unblock@1.0.0
└─┬ lamhieu:meteorx@2.1.1

---------

meteor run     
[[[[[ ~/Development/payagol ]]]]]             

=> Started proxy.                             
=> Started MongoDB.                           
W20190529-14:25:59.624(-7)? (STDERR) /Users/tom/.meteor/packages/meteor-tool/.1.8.1.7c0itz.u7ktr++os.osx.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.osx.x86_64/dev_bundle/server-lib/node_modules/fibers/future.js:280
W20190529-14:25:59.670(-7)? (STDERR) 						throw(ex);
W20190529-14:25:59.670(-7)? (STDERR) 						^
W20190529-14:25:59.670(-7)? (STDERR) 
W20190529-14:25:59.671(-7)? (STDERR) TypeError: Cannot read property 'constructor' of undefined
W20190529-14:25:59.671(-7)? (STDERR)     at exposeSubscription (packages/meteorhacks_meteorx.js:61:41)
W20190529-14:25:59.671(-7)? (STDERR)     at exposeLivedata (packages/meteorhacks_meteorx.js:31:5)
W20190529-14:25:59.671(-7)? (STDERR)     at packages/meteorhacks_meteorx.js:175:1
W20190529-14:25:59.671(-7)? (STDERR)     at packages/meteorhacks_meteorx.js:200:4
W20190529-14:25:59.671(-7)? (STDERR)     at packages/meteorhacks_meteorx.js:208:3
W20190529-14:25:59.672(-7)? (STDERR)     at /Users/tom/Development/payagol/.meteor/local/build/programs/server/boot.js:419:36
W20190529-14:25:59.672(-7)? (STDERR)     at Array.forEach (<anonymous>)
W20190529-14:25:59.672(-7)? (STDERR)     at /Users/tom/Development/payagol/.meteor/local/build/programs/server/boot.js:228:19
W20190529-14:25:59.672(-7)? (STDERR)     at /Users/tom/Development/payagol/.meteor/local/build/programs/server/boot.js:479:5
W20190529-14:25:59.672(-7)? (STDERR)     at Function.run (/Users/tom/Development/payagol/.meteor/local/build/programs/server/profile.js:510:12)
W20190529-14:25:59.672(-7)? (STDERR)     at /Users/tom/Development/payagol/.meteor/local/build/programs/server/boot.js:478:11

**Comment by adamgins** [comment]
Can't you just use https://docs.meteor.com/api/methods.html#DDPCommon-MethodInvocation-unblock now?

**Comment by FinnFrotscher** [comment]
+1. replaced meteorhacks:unblock with lamhieu:unblock. still nothing

**Comment by afrokick** [comment]
@FinnFrotscher you should update all packages which contain `meteorhacks:meteorx` in deps. Could you please share the output of `meteor list --tree`?
